builder/proxmox/common/step_download_iso_on_pve.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package proxmox
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"path"
+
+	"github.com/Telmate/proxmox-api-go/proxmox"
+	"github.com/hashicorp/go-getter/v2"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// stepDownloadISOOnPVE downloads an ISO file directly to the specified PVE node.
+// Checksums are also calculated and compared on the PVE node, not by Packer.
+type stepDownloadISOOnPVE struct {
+	ISO *additionalISOsConfig
+}
+
+func (s *stepDownloadISOOnPVE) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	var isoStoragePath string
+	isoStoragePath = Download_iso_on_pve(state, s.ISO.ISOUrls, s.ISO.ISOChecksum, s.ISO.ISOStoragePool)
+
+	if isoStoragePath != "" {
+		s.ISO.ISOFile = isoStoragePath
+		return multistep.ActionContinue
+	}
+	// Abort if no ISO can be downloaded
+	return multistep.ActionHalt
+}
+
+func Download_iso_on_pve(state multistep.StateBag, ISOUrls []string, ISOChecksum string, ISOStoragePool string) string {
```

Comment:
It wasn't immediately clear to me  what this was returning. I recommend documenting this function to indicate what it does.

Reply:
You are right. I will add some comments to make it more clear.

Reply:
In terms of code style, Go encourages usage of camelCase or PascalCase for function names depending on visibility, so I would suggest we rename this to `DownloadISOOnPVE` in order to be more in-line with the guidelines for the language.